harness/determined/_core/_checkpoint.py

```diff
@@ -106,44 +142,70 @@ def store_path(
 
         .. code::
 
-           with checkpointing.store_path() as (path, storage_id):
+           with core_context.checkpoint.store_path() as (path, storage_id):
                my_save_model(my_model, path)
                print(f"done saving checkpoint {storage_id}")
            print(f"done uploading checkpoint {storage_id}")
         """
 
         if self._dist.rank != 0:
-            raise ValueError(
-                "cannot call checkpointing.store_path() from non-chief worker "
+            raise RuntimeError(
+                "cannot call CheckpointContext.store_path() from non-chief worker "
                 f"(rank={self._dist.rank})"
             )
 
         storage_id = str(uuid.uuid4())
         with self._storage_manager.store_path(storage_id) as path:
             yield path, storage_id
-            resources = storage.StorageManager._list_directory(path)
+            resources = self._storage_manager._list_directory(path)
```

Comment:
isn't `_list_directory` a static method? If yes isn't the existing call preferable?

Reply:
oh, is it for mockability??

Reply:
it is a static method.  Getting it from the storage_manager lets the storage_manager instance override the list_directory via class inheritance, while `storage.StorageManager._list_directory` is referring to a specific class's implementation.

I would argue that if you don't want to use the instances' _list_directory, I'd argue that it shouldn't be a staticmethod at all, but a helper function in the storage module.

Reply:
👍